committer/src/config.rs

```diff
@@ -35,8 +35,35 @@ pub struct EthConfig {
     /// Ethereum address of the fuel chain state contract.
     pub state_contract_address: Address,
     /// The number of fuel blocks between ethereum commits. If set to 1, then every block should be pushed to Ethereum.
+    #[serde(
+        default = "default_commit_interval",
+        deserialize_with = "deserialize_commit_interval"
+    )]
     pub commit_interval: NonZeroU32,
```

Comment:
Two things:
1. If the only source of truth is what the contract says, I'd opt for not configuring the interval at all. i.e. removing it from the config. Also it is kinda misleading, allowing ppl to configure it only to have it be overwritten later on.
2. I dislike having a default that can kick in for prod for such a sensitive thing.